#thor:
# trajectory controllers ---------------------------------------
 left_arm_traj_controller:
    type: "position_controllers/JointTrajectoryController"
    joints:
      - l_shoulder_pitch
      - l_shoulder_roll
      - l_shoulder_yaw
      - l_elbow
      - l_wrist_yaw1
      - l_wrist_roll
      - l_wrist_yaw2
      
    constraints:
      goal_time: &goal_time_constraint 4.0
      stopped_velocity_tolerance: 1.0
      l_shoulder_pitch:
        goal: &goal_pos_constraint 0.5
        trajectory: &trajectory_pos_constraint 1.0
      l_shoulder_roll:
        goal: *goal_pos_constraint
        trajectory: *trajectory_pos_constraint        
      l_shoulder_yaw:
        goal: *goal_pos_constraint
        trajectory: *trajectory_pos_constraint
      l_elbow:
        goal: *goal_pos_constraint
        trajectory: *trajectory_pos_constraint        
      l_wrist_yaw1:
        goal: *goal_pos_constraint
        trajectory: *trajectory_pos_constraint
      l_wrist_roll:
        goal: *goal_pos_constraint
        trajectory: *trajectory_pos_constraint
      l_wrist_yaw2:
        goal: *goal_pos_constraint
        trajectory: *trajectory_pos_constraint
        
        
 right_arm_traj_controller:
    type: "position_controllers/JointTrajectoryController"
    joints:
      - r_shoulder_pitch
      - r_shoulder_roll
      - r_shoulder_yaw
      - r_elbow
      - r_wrist_yaw1
      - r_wrist_roll
      - r_wrist_yaw2
      
    constraints:
      goal_time: *goal_time_constraint
      stopped_velocity_tolerance: 1.0
      r_shoulder_pitch:
        goal: *goal_pos_constraint
        trajectory: *trajectory_pos_constraint
      r_shoulder_roll:
        goal: *goal_pos_constraint
        trajectory: *trajectory_pos_constraint
      r_shoulder_yaw:
        goal: *goal_pos_constraint
        trajectory: *trajectory_pos_constraint
      r_elbow:
        goal: *goal_pos_constraint
        trajectory: *trajectory_pos_constraint        
      r_wrist_yaw1:
        goal: *goal_pos_constraint
        trajectory: *trajectory_pos_constraint
      r_wrist_roll:
        goal: *goal_pos_constraint
        trajectory: *trajectory_pos_constraint
      r_wrist_yaw2:
        goal: *goal_pos_constraint
        trajectory: *trajectory_pos_constraint
   

 left_leg_traj_controller:
    type: "position_controllers/JointTrajectoryController"
    joints:
      - l_hip_yaw
      - l_hip_roll
      - l_hip_pitch
      - l_knee
      - l_ankle_pitch
      - l_ankle_roll
      
    constraints:
      goal_time: *goal_time_constraint
      stopped_velocity_tolerance: 1.0
      l_hip_yaw:
        goal: *goal_pos_constraint
        trajectory: *trajectory_pos_constraint
      l_hip_roll:
        goal: *goal_pos_constraint
        trajectory: *trajectory_pos_constraint        
      l_hip_pitch:
        goal: *goal_pos_constraint
        trajectory: *trajectory_pos_constraint
      l_knee:
        goal: *goal_pos_constraint
        trajectory: *trajectory_pos_constraint        
      l_ankle_pitch:
        goal: *goal_pos_constraint
        trajectory: *trajectory_pos_constraint
      l_ankle_roll:
        goal: *goal_pos_constraint
        trajectory: *trajectory_pos_constraint
   

 right_leg_traj_controller:
    type: "position_controllers/JointTrajectoryController"
    joints:
      - r_hip_yaw
      - r_hip_roll
      - r_hip_pitch
      - r_knee
      - r_ankle_pitch
      - r_ankle_roll
      
    constraints:
      goal_time: *goal_time_constraint
      stopped_velocity_tolerance: 1.0
      r_hip_yaw:
        goal: *goal_pos_constraint
        trajectory: *trajectory_pos_constraint
      r_hip_roll:
        goal: *goal_pos_constraint
        trajectory: *trajectory_pos_constraint        
      r_hip_pitch:
        goal: *goal_pos_constraint
        trajectory: *trajectory_pos_constraint
      r_knee:
        goal: *goal_pos_constraint
        trajectory: *trajectory_pos_constraint        
      r_ankle_pitch:
        goal: *goal_pos_constraint
        trajectory: *trajectory_pos_constraint
      r_ankle_roll:
        goal: *goal_pos_constraint
        trajectory: *trajectory_pos_constraint


 torso_traj_controller:
    type: "position_controllers/JointTrajectoryController"
    joints:
      - waist_pan
      - waist_tilt
      
    constraints:
      goal_time: *goal_time_constraint
      stopped_velocity_tolerance: 1.0
      waist_pan:
        goal: *goal_pos_constraint
        trajectory: *trajectory_pos_constraint
      waist_tilt:
        goal: *goal_pos_constraint
        trajectory: *trajectory_pos_constraint
        
 head_traj_controller:
    type: "position_controllers/JointTrajectoryController"
    joints:
      - head_pan
      - head_tilt
      
    constraints:
      goal_time: *goal_time_constraint
      head_pan:
        goal: *goal_pos_constraint
        trajectory: *trajectory_pos_constraint
      head_tilt:
        goal: *goal_pos_constraint
        trajectory: *trajectory_pos_constraint        
        
 waist_lidar_controller:          
    type: "position_controllers/JointTrajectoryController"
    joints:
      - waist_lidar
      
    constraints:
      goal_time: *goal_time_constraint
      waist_lidar:
        goal: *goal_pos_constraint
        
    stop_trajectory_duration: 1.0   
